src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -2853,8 +2859,23 @@ string ApplyCast(string value, bool useSafeCast, params XamlObjectDefinition[] s
                 {
                     return $"({type}){value}";
                 }
-            }
-        }
+			}
+
+			// this allows for the resource to be implicitly converted to another type, however
+			// its implementation may not be implicit
+			string TryImplicitXamlConvert(string value, XamlObjectDefinition sourceType)
+			{
+				// note: Source is a non-prefixed xaml name
+				switch ((Source: sourceType.Type.Name, Target: targetPropertyType.ToDisplayString()))
```

Comment:
Another way to write `switch` statements that only have one case is a little-known syntax called the `if` statement :p